lib/modules/manager/mise/extract.spec.ts

```diff
@@ -0,0 +1,326 @@
+      [tools]
+      erlang = ['23.3', '24.0']
+      node = ['16', 'prefix:20', 'ref:master', 'path:~/.nodes/14']
+    `;
+      const result = extractPackageFile(content, miseFilename);
+      expect(result).toMatchObject({
+        deps: [
+          {
+            depName: 'erlang',
+            currentValue: '23.3',
+            datasource: 'github-tags',
+          },
+          {
+            depName: 'node',
+            currentValue: '16',
+            datasource: 'node-version',
+          },
+        ],
+      });
+    });
+
+    it('extracts tools with plugin options', () => {
+      const content = codeBlock`
+      [tools]
+      python = {version='3.11', virtualenv='.venv'}
+    `;
+      const result = extractPackageFile(content, miseFilename);
+      expect(result).toMatchObject({
+        deps: [
+          {
+            depName: 'python',
+            currentValue: '3.11',
+          },
+        ],
+      });
+    });
+
+    it('provides skipReason for lines with unsupported tooling', () => {
+      const content = codeBlock`
+      [tools]
+      fake-tool = '1.0.0'
+    `;
+      const result = extractPackageFile(content, miseFilename);
+      expect(result).toMatchObject({
+        deps: [
+          {
+            depName: 'fake-tool',
+            skipReason: 'unsupported-datasource',
+          },
+        ],
+      });
+    });
+
+    it('provides skipReason for missing version - empty string', () => {
+      const content = codeBlock`
+      [tools]
+      python = ''
+    `;
+      const result = extractPackageFile(content, miseFilename);
+      expect(result).toMatchObject({
+        deps: [
+          {
+            depName: 'python',
+            skipReason: 'unspecified-version',
+          },
+        ],
+      });
+    });
+
+    it('provides skipReason for missing version - missing version in object', () => {
+      const content = codeBlock`
+      [tools]
+      python = {virtualenv='.venv'}
+    `;
+      const result = extractPackageFile(content, miseFilename);
+      expect(result).toMatchObject({
+        deps: [
+          {
+            depName: 'python',
+            skipReason: 'unspecified-version',
+          },
+        ],
+      });
+    });
+
+    it('complete .mise.toml example', () => {
+      const result = extractPackageFile(mise1toml, miseFilename);
+      expect(result).toMatchObject({
+        deps: [
+          {
+            depName: 'java',
+            currentValue: '21.0.2',
+            datasource: 'java-version',
+          },
+          {
+            depName: 'erlang',
+            currentValue: '23.3',
+            datasource: 'github-tags',
+          },
+          {
+            depName: 'node',
+            currentValue: '16',
+            datasource: 'node-version',
+          },
+        ],
+      });
+    });
+
+    it('complete example with skip', () => {
+      const content = codeBlock`
+      [tools]
+      java = '21.0.2'
+      erlang = ['23.3', '24.0']
```

Comment:
```suggestion
      erlang = []
```
Add a test that empty arrays are handled gracefully.